src/Persistence/Array_.php

```diff
@@ -397,27 +354,33 @@ public function action($m, $type, $args = [])
 
                 $field = is_string($args[0]) ? $args[0] : $args[0][0];
 
-                $action = $this->initAction($m, [$field]);
-                $this->applyConditions($m, $action);
-                $this->setLimitOrder($m, $action);
+                $action = $this->initAction($model, [$field]);
+                $this->applyScope($model, $action);
+                $this->setLimitOrder($model, $action);
 
                 // get first record
-                $row = $action->getRow();
-                if ($row) {
+                if ($row = $action->getRow()) {
                     if (isset($args['alias']) && array_key_exists($field, $row)) {
                         $row[$args['alias']] = $row[$field];
                         unset($row[$field]);
                     }
                 }
 
                 return $row;
-            /* These are not implemented yet
             case 'fx':
             case 'fx0':
+                if (!isset($args[0], $args[1])) {
+                    throw (new Exception('fx action needs 2 arguments, eg: ["sum", "amount"]'))
+                        ->addMoreInfo('action', $type);
+                }
 
-                return $action->aggregate($field->short_name, $fx);
-            */
+                $fx = $args[0];
+                $field = $args[1] ?? null;
```

Comment:
never null, see condition above... I also prefer not to combine `isset`, harder to read

Reply:
done